### ISSUE TITLE
Include Display Settings Menu 

### ISSUE THREAD

**Issue by DanKothlow** [Thinstation/thinstation#869]
Good afternoon.
We are using ThinStation with FreeRDP, and have that portion worked out well. 
We are trying to have a setup with ICONMODE=MANUAL to hide all the start menu options, but also include the Display settings menu option, to account for different monitor configurations and display resolutions.

Is that possible?

Thank you,
Dan Kothlow

**Comment by Doncuppjr** [comment]
Yes, but you would have to back-fill some ThinStation specific constructs that call xfce4-display-settings through the thinstation.packages vs directly. You could then define a session for xfce4-display-settings with an ICON=true and it would appear on the desktop. Alternatively, you could hack on packages/xfwm4/etc/xfwm.functions to generate the menu entry always. If you come up with some good code, make a pull request, it might be useful to others.

**Comment by Doncuppjr** [comment]
Yes, but you would have to back-fill some ThinStation specific constructs that call xfce4-display-settings through the thinstation.packages vs directly. You could then define a session for xfce4-display-settings with an ICON=MENU and it would appear in the start menu. Alternatively, you could hack on packages/xfwm4/etc/xfwm.functions to generate the menu entry always. If you come up with some good code, make a pull request, it might be useful to others.

**Comment by DanKothlow** [comment]
I'm at a bit of a loss on how to do the construct through thinstation.packages. 
Could you possibly direct me to an example of that?
And would the session be similar to how we do the session for freerdp settings?

Thank you.

**Comment by Doncuppjr** [comment]
Here is a page that goes over the complete possible elements to a package. https://github.com/Thinstation/thinstation/wiki/Packaging
You'll want the last part of Initialization and Startup to complete the elements you need.
Yes, it would then be like Freerdp, except set ICON=MENU to put it in the menu.

**Comment by DanKothlow** [comment]
I am still at a loss on how this would work. 
I looked quite a bit through the /build folder for "xfce4-display-settings". 
I can't find anything about it.

**Comment by Doncuppjr** [comment]
It's a binary and located in the xfwm4 package. It's icon is currently only generated in ICONMODE=AUTO mode, which is the default. In order for it to show up in ICONMODE=MANUAL, you would have to at least come up with a name you want to use, then make a link in etc/init.d of that name that points to /etc/thinstation.packages. Create a global file in etc/cmd that calls that binary, then make a session for it in your configs. You can make your own customizations package that contains just those two artifacts and include it in your build, instead of editing xfwm4 directly. Then it's portable and you don't have to worry about always recreating them.

**Comment by Doncuppjr** [comment]
If your just trying to remove unwanted icons, it might be easier to just remove /lib/menu files that you don't want to show up.

**Comment by DanKothlow** [comment]
We are wanting to essentially run KIOSK mode with Freerdp, but also need to have the ability to customize the display settings, as noted in the original post. 

Here's what I tried. 
In the build.conf file, I added a line "package display_setttings"
In build/packages, I created a folder called "display_settings"
In that folder, I created 2 folders - "etc" and "cmd".
In the cmd folder, I created a file called "display_settings.global"
In that file, I have "CMD_GLOBAL="xfce4-display-settings"

In the etc folder, I created a folder called "init.d".
In that folder, I crated a link called "display_setttings" that points to /etc/thinstation.packages

In the thinstation.conf.buildtime I have a line that reads "SESSION_1_ICON=MENU"

It does not put the display settings option on the menu. 
Do I need to reference the global variable somehow?

**Comment by Doncuppjr** [comment]
I would guess that you also need
SESSION_1_TYPE="display_settings"
SESSION_1_TITLE="Display Settings"
SESSION_1_ICON=ON #Put it on the desktop since KIOSK mode has no menu.

display_settings.global show go in /etc/cmd/.

**Comment by DanKothlow** [comment]
I realized about /etc/cmd after I didn't see the file on the target machine. 

After addding - 
SESSION_1_TYPE="display_settings"
SESSION_1_TITLE="Display Settings"
SESSION_1_ICON=ON

I get an error "Package display_settings doesn't exist aborting bootup.

Before I added those 3 lines and the system would boot, the package files were in /etc/init.d and /etc/cmd.

**Comment by Doncuppjr** [comment]
Something is wrong with your /etc/init.d/ link. That error only show up if you have a mismatch between the SESSION TYPE and the name of the link, a missing link, or the link is pointing to the wrong place.

**Comment by DanKothlow** [comment]
I had "display_setttings" - 3 t 

I have Freerdp as SESSION_1 and display_settings as SESSION_2.
I have Freerdp autostart on, and display_settings autostart off. 

It's working just as we expected. 

Thank you for your assistance.
